### PR TITLE
Signup with facebook #8

### PR DESCRIPTION
**Pull request by jass2**

### CODE REVIEW

File: app/views/devise/shared/_links.html.erb [hunk 1]
```diff
@@ -19,9 +19,10 @@
     <%= link_to t('.didn_t_receive_unlock_instructions'), new_unlock_path(resource_name) %><br />
   <% end -%>
 
-  <%- if devise_mapping.omniauthable? %>
+  <%- if devise_mapping.omniauthable? && AppConfig.facebook_login_enabled %>
```
Comment (dfurber) [inline app/views/devise/shared/_links.html.erb]: This is looking good. Next step is to add feature specs to prove that:
1. The FB link shows when enabled
2. The FB link doesn't show when not enabled
3. That the FB login works - both to add to existing user and to create a new user, if creating new users is enabled

This looks promising as a place to start: https://stackoverflow.com/questions/23050989/how-to-test-facebook-login-action-with-rails-omniauth-and-rspec

**Review by dfurber — CHANGES_REQUESTED** [review]
Now for some tests...

File: app/views/devise/shared/_links.html.erb (hunk 1 above)
Comment (jass2) [reply app/views/devise/shared/_links.html.erb]: Let me know how these look: https://github.com/historyforge/historyforge/pull/59/commits/e91fe28d6d28a263250971c4d653e1d547f01567

**Comment by dfurber** [comment]
Bet you could remove the Devise mappings omniauthable part because we know
it is.

File: db/seeds.rb [hunk 2]
```diff
@@ -46,6 +46,9 @@
 Setting.add 'geocoding_key', value: ENV['GOOGLE_GEOCODING_KEY'], group: 'API Keys'
 Setting.add 'recaptcha_site_key', value: nil, group: 'API Keys'
 Setting.add 'recaptcha_secret_key', value: nil, group: 'API Keys'
+Setting.add 'facebook_login_enabled', type: :boolean, value: ENV['FACEBOOK_LOGIN_ENABLED'] || '0', group: 'API Keys', name: 'Enable Facebook Authentication', hint: 'Allow users to signup with their Facebook account.'
```
Comment (dfurber) [inline db/seeds.rb]: Seems to want its own group?

**Pull request merged**